### PR TITLE
Fix up some button things

### LINKED ISSUE #127: Incorrect button values
Buttons don't have any transition easing and the padding sizes are incorrect.

## Expected behavior
Hover should have at least a `transition: all .25s ease-in-out;`
[Example provided with correct padding and transition](https://codepen.io/levinriegner/pen/b442b81d8b6fcefe7e38b61d35eeddac?editors=1100)

## Current behavior
Padding doesnt have enough top/bottom.
Screenshot 2021-04-22 at 14 42 50

Hover is an [instant transition](https://www.loom.com/share/d89fa73fa8424e08b3bf76754a82d84d)

## Steps to reproduce
1. Go to [the website](https://www.pulumi.com/)
2. Hover over any purple button

## Affected feature
Primary Buttons

### PR DESCRIPTION
**Pull request by cnunciato**
* Make a `gradient-button` mixin with support for transitions
* Apply some styles to the buttons in the `learnmore-contactus` partial
* Add a bit of margin between the Download button icon and label
* Extend the default transition duration to 120ms
* Remove some unused styles
* Remove some no-longer-needed punctuation

fixes #127

### CODE REVIEW

File: themes/default/assets/sass/_buttons.scss [hunk 1]
```diff
@@ -41,30 +40,6 @@
...
-}
-
-.btn-white {
-    @apply bg-white text-gray-900 border-white;
-
-    &:hover {
-        @apply bg-gray-300 text-gray-900;
-    }
-}
-
```
Comment (cnunciato) [inline themes/default/assets/sass/_buttons.scss:67]: These are no longer used, so we can remove them.

File: themes/default/assets/sass/_buttons.scss [hunk 2]
```diff
@@ -91,6 +66,10 @@
     &:hover {
         @apply border-gray-700 text-gray-700 bg-transparent;
     }
+
+    i {
+        @apply mr-1;
+    }
```
Comment (cnunciato) [inline themes/default/assets/sass/_buttons.scss:72]: Just adds a bit of breathing room for the buttons [on this page](https://www.pulumi.com/brand/).

**Review by susanev — APPROVED** [review]

**Comment by cnunciato** [comment]
Looks like I have one little thing to fix up: on the Solutions page, on hover, the z-index is too far back:

image

Will fix that up and merge on Monday.

**Comment by cnunciato** [comment]
Ok, I have all the things working now, with the transitions and everything, so going to merge.

**Pull request merged**